rust/cubesql/cubesql/src/compile/engine/information_schema/postgres/pg_collation.rs

```diff
@@ -0,0 +1,259 @@
+use std::sync::Arc;
+
+use async_trait::async_trait;
+
+use datafusion::{
+    arrow::{
+        array::{Array, ArrayRef, BooleanBuilder, Int32Builder, StringBuilder, UInt32Builder},
+        datatypes::{DataType, Field, Schema},
+        record_batch::RecordBatch,
+    },
+    datasource::{datasource::TableProviderFilterPushDown, TableProvider},
+    error::Result,
+    logical_plan::Expr,
+    physical_plan::{memory::MemoryExec, ExecutionPlan},
+};
+
+use crate::compile::engine::information_schema::postgres::PG_NAMESPACE_CATALOG_OID;
+
+struct PgCollation {
+    oid: u32,
+    collname: &'static str,
+    collnamespace: u32,
+    collowner: u32,
+    collprovider: String,
+    collisdeterministic: bool,
+    collencoding: i32,
+    collcollate: Option<String>,
+    collctype: Option<String>,
+    // Column `colliculocale` renamed to `colllocale` since PostgreSQL 17.
+    // Support both columns for backward-compatibility.
+    // Reference: https://pgpedia.info/p/pg_collation.html
+    colliculocale: Option<String>,
+    colllocale: Option<String>,
```

Comment:
Since `colllocale` is simply renamed `colliculocale`, there's no reason to have both in `PgCollation`. It's best to remove one of them and use it to fill both fields in the builder.

```suggestion
    // Column `colliculocale` renamed to `colllocale` since PostgreSQL 17.
    colllocale: Option<String>,
```